### PR TITLE
Implement kubeconfig rotation

### PR DESCRIPTION
**Pull request by Disper**
**Description**

Changes proposed in this pull request:

- Periodic secret rotation implemented
- Forced secret rotation implemented
- Statuses set for operations

**Related issue(s)**
#10

### CODE REVIEW

File: internal/controller/gardener_cluster_controller.go [hunk 1]
```diff
@@ -162,16 +203,146 @@ func (r *GardenerClusterController) getSecret(shootName string) (*corev1.Secret,
...
+
+	lastSyncTimeString := annotations[lastKubeconfigSyncAnnotation]
+	lastSyncTime, err := time.Parse(time.RFC3339, lastSyncTimeString)
+	if err != nil {
+		return true
+	}
+	now := time.Now()
+	alreadyValidFor := now.Sub(lastSyncTime)
+
+	return alreadyValidFor.Minutes() >= rotationPeriodRatio*rotationPeriod.Minutes()
```
Comment (Disper) [inline internal/controller/gardener_cluster_controller.go:264]: would it be possible to remove this rotationPeriodRatio parameter?

File: internal/controller/gardener_cluster_controller_test.go [hunk 2]
```diff
@@ -15,49 +15,195 @@ import (
...
+			gardenerClusterKey := types.NamespacedName{Name: gardenerClusterCR.Name, Namespace: gardenerClusterCR.Namespace}
+			var newGardenerCluster imv1.GardenerCluster
+
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), gardenerClusterKey, &newGardenerCluster)
+				if err != nil {
+					return false
+				}
+
+				readyState := newGardenerCluster.Status.State == imv1.ReadyState
```
Comment (Disper) [inline internal/controller/gardener_cluster_controller_test.go:144]: Happened to failed for @akgalwas locally

**Comment by akgalwas** [comment]
What should be taken into account during review:
- Errors handling
- Resilience: we must be sure that the controller is self healing
- Unit tests should be reviewed
- Unit test stability (it would be good to run it couple of times)
- Force rotation scenario: there is update failure caused by the Kubebuilder cache

File: internal/controller/gardener_cluster_controller.go (hunk 1 above)
Comment (Disper) [reply internal/controller/gardener_cluster_controller.go:264]: without that, @akgalwas had a test case where instead after 36 minutes, rotation happened after ~72 minutes and if I recall correctly, for 12 minutes the kubeconfig was invalid.

**Review by m00g3n — APPROVED** [review]
LGTM

**Pull request merged**